apisix/plugins/hmac-auth.lua

```diff
@@ -327,6 +351,18 @@ local function validate(ctx, params)
         return nil, {message = "Invalid signature"}
     end
 
+    local validate_request_body = get_conf_field(params.access_key, "validate_request_body")
+    if validate_request_body then
+        local max_req_body = get_conf_field(params.access_key, "max_req_body")
```

Comment:
I think we could return here if no params.body_digest.

Reply:
I think we should check whether the  params.body_digest exists or not.

Reply:
when validate_request_body == true and no params.body_digest, why not return false?

Reply:
When there is no request body, the `X-HMAC-DIGEST` header can be omitted.

Reply:
@arthur-zhang 

OK. Thanks for your explanation. How about when `validate_request_body` is enabled, `X-APISIX-HMAC-BODY-DIGEST` is required? Just pass empty strings for it when no body.

Reply:
Fix it